File: derivatives.exs
def derive() do
  :ok
end


def derive(term, "") do
  #look for power
  #look for number in front of power
  #multiplying the number and the power
  #reducing the power by 1
end


def get_derivative(term) do
  derive(term, "")
end


term = IO.gets "Geben Sie den Term ein, den Sie ableiten wollen: "
dNum = IO.gets "Zu welchem Grad wollen Sie ableiten? "
IO.puts "#{term} und #{dNum}"
get_derivative(term, dNum)

File: derivatives.py
def get_subterms(term):
    subterms = []
    searching = True
    seperatorPositions = []
    while searching:
        seperatorPos = term.find('+')
        if seperatorPos == -1:
            seperatorPos = term.find('-')
            if seperatorPos == -1:
                searching = False
        else:
            seperatorPositions.append(seperatorPos)
    prevPos = 0
    for pos in seperatorPositions:
        subterms.append(term[prevPos:pos])
        prevPos = pos
    return subterms 

def derivate(term):
    pass


def get_derivative(term, dNum):
    result = ""
    for i in range(dNum):
        get_subterms(term)


term = input("Welchen Term wollen Sie ableiten: ")
dNum = input("Welchen Grad soll die Ableitung haben: ")
